sql/testdata/subquery

```diff
@@ -135,3 +135,8 @@ query IT
 SELECT * FROM kv WHERE k = (SELECT k FROM kv WHERE (k, v) = (1, 'one'))
 ----
 1 one
+
+query B
```

Comment:
I wasn't planning on merging this; this branch is on the main repo so that someone (or I) can replace my commit with one that contains a fix.